### PR TITLE
implement key-mixin

### PR DESCRIPTION
**Pull request by nelix**

### CODE REVIEW

File: packages/key-mixin/example.jsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  },
+
+  handleEnter(e) {
+    alert('enter');
+  },
+
+  render() {
+    return (
+      <input
+        {...this.callOnKeyDown(hotKeys)}
```
Comment (grapegravity) [inline packages/key-mixin/example.jsx]: Is using spread syntax here really giving us that much?

Comment (nelix) [inline packages/key-mixin/example.jsx]: mostly just playing around with the idea, I feel it helps separate (obfuscate!?) the concern of the mixin from the component that is using it. It allows us to change the events, or hook multiple events on a component for more complicated hot key handling  in the future... but in this case it maybe

File: packages/key-mixin/index.jsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      ) && value.cb
+    );
+    callbacks = _.compact(callbacks);
+
+    _.each(callbacks, (cb) => this[cb](e));
+
+    if (callbacks.length > 0) return false;
+  },
+
+  bindKeys(keyMask) {
```
Comment (grapegravity) [inline packages/key-mixin/index.jsx]: I’d consider renaming this to something like `keyDownHandler`, as `bindKeys` isn’t really what it does functionally so much as programmatically. :wink:

Comment (nelix) [inline packages/key-mixin/index.jsx]: its no longer keyDown specific, so... keyHandler?

Comment (nelix) [inline packages/key-mixin/index.jsx]: I based the name on fluxxor's bindActions

File: packages/key-mixin/example.jsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  handleEnter(e) {
+    alert('enter');
+  },
+
+  render() {
+    return (
+      <input
+        onKeyDown={this.keyHandler(keys)}
+      />
```
Comment (grapegravity) [inline packages/key-mixin/example.jsx]: I’d add some instructional text around the example, so the user knows to press escape or enter.

File: packages/key-mixin/example.jsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  displayName: 'KeyMixinExample',
+  mixins: [KeyMixin],
+
+  handleEscape(e) {
+    alert('escape');
+  },
+
+  handleEnter(e) {
+    alert('enter');
+  },
```
Comment (grapegravity) [inline packages/key-mixin/example.jsx]: Can we output this under the input rather than with an alert?

Comment (nelix) [inline packages/key-mixin/example.jsx]: i wanted to keep the example short and specific

Comment (grapegravity) [inline packages/key-mixin/example.jsx]: `alert`s are kinda nasty, though

**Pull request merged**